src/main.rs

```diff
@@ -113,16 +119,10 @@ fn main() {
                         action_stack.push(UserAction::IntroParsed);
                     }
                     UserAction::RemoveInput => {
-                        let dead_node_name = other.0.as_str();
-                        let dead_node = &user_data
-                            .inputs
-                            .clone()
-                            .unwrap()
-                            .get(dead_node_name)
-                            .unwrap()
-                            .parent()
-                            .unwrap();
-                        let new_root = match kill_node_attribute(dead_node) {
+                        let dead_node_path = other.0.as_str();
+                        let inputs = user_data.inputs.clone().unwrap();
```

Comment:
why is that `.clone()` needed and `.as_ref()` doesn't suffice?

Reply:
If I use `as_ref()` here, I think I'm indirectly getting an immutable reference to`user_data`. Later on, I'm overwriting `user_data` with `new_root` on line 138, which causes an error since I've also got an immutable reference.